packages/reactivity/src/reactive.ts

```diff
@@ -65,26 +65,33 @@ function getTargetType(value: Target) {
 export type UnwrapNestedRefs<T> = T extends Ref ? T : UnwrapRefSimple<T>
 
 /**
- * Creates a reactive copy of the original object.
+ * Creates a deeply-reactive copy of the original object.
  *
- * The reactive conversion is "deep"—it affects all nested properties. In the
- * ES2015 Proxy based implementation, the returned proxy is **not** equal to the
- * original object. It is recommended to work exclusively with the reactive
- * proxy and avoid relying on the original object.
+ * All (deep) refs within the object will be unwrapped, while maintaining the
+ * ref's reactivity. This is also done for refs that are assigned after creation
+ * of the proxy object.
  *
- * A reactive object also automatically unwraps refs contained in it, so you
- * don't need to use `.value` when accessing and mutating their value:
+ * Note that the proxy object is not equal to the original object in the ES2015
+ * Proxy-based implementation. It is **strongly** recommended to work
+ * exclusively with the reactive proxy and avoid relying on the original object.
```

Comment:
I think it might be worth shortening the description and the example a bit and add a link to the docs for further details: https://vuejs.org/api/reactivity-core.html#reactive
But let's wait on more feedback for that